creusot/src/backend/dependency.rs

```diff
@@ -23,7 +23,10 @@ use super::{ty_inv::tyinv_head_and_subst, PreludeModule, TransId};
 pub(crate) enum Dependency<'tcx> {
     Type(Ty<'tcx>),
     Item(DefId, GenericArgsRef<'tcx>),
+    // Type invariants and structual resolution expressions
+    // are identified by a "skeleton type" aka type with identity substs.
```

Comment:
In `Dependency`, we use the full type. It's in `TransId` that we use the skeletton type.